lib/Benchmark/Executor/HrtimeExecutor.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the PHPBench package
+ *
+ * (c) Daniel Leech <daniel@dantleech.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ */
+
+namespace PhpBench\Benchmark\Executor;
+
+use PhpBench\Benchmark\Remote\Payload;
+use PhpBench\Model\Iteration;
+use PhpBench\Model\Result\MemoryResult;
+use PhpBench\Model\Result\TimeResult;
+use PhpBench\Registry\Config;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This class generates a benchmarking script and places it in the systems
+ * temp. directory and then executes it. The generated script then returns the
+ * time taken to execute the benchmark and the memory consumed.
+ */
+class HrtimeExecutor extends BaseExecutor
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function launch(Payload $payload, Iteration $iteration, Config $options)
+    {
+        $phpConfig = [
+            'max_execution_time' => 0,
+        ];
+
+        $payload->mergePhpConfig($phpConfig);
+        $result = $payload->launch(__DIR__ . '/template/hrtime.template');
+
+        if (isset($result['buffer']) && $result['buffer']) {
+            throw new \RuntimeException(sprintf(
+                'Benchmark made some noise: %s',
+                $result['buffer']
+            ));
+        }
+
+        $iteration->setResult(new TimeResult($result['time']));
```

Comment:
Is `time` still microseconds? I wonder if we should either multiply the HR time to microseconds or specify the unit here (potentially even a new `Result` class).

Reply:
hrtime returns nanoseconds currently I calculate the microseconds but would be better to keep full accuracy. Also currently I use numeric result which could loose accuracy on 32bit platforms.An alternative would be to return an array of seconds and nanoseconds.